extensions/wsc/src/main/java/org/mule/extension/ws/internal/introspection/WsdlIntrospecter.java

```diff
@@ -0,0 +1,171 @@
+    return port;
+  }
+
+  public List<String> getOperationNames() {
+    List<BindingOperation> bindingOperations = (List<BindingOperation>) port.getBinding().getBindingOperations();
+    return bindingOperations.stream().map(BindingOperation::getName).collect(Collectors.toList());
+  }
+
+  public Operation getOperation(String operationName) {
+    validateBlankString(operationName, "operation name");
+    Operation operation = port.getBinding().getPortType().getOperation(operationName, null, null);
+    validateNotNull(operation, "The operation name [" + operationName + "] was not found in the current wsdl file.");
+    return operation;
+  }
+
+  public BindingOperation getBindingOperation(String operationName) {
+    validateBlankString(operationName, "operation name");
+    BindingOperation operation = port.getBinding().getBindingOperation(operationName, null, null);
+    validateNotNull(operation, "The binding operation name [" + operationName + "] was not found in the current wsdl file.");
+    return operation;
+  }
+
+  public Fault getFault(Operation operation, String faultName) {
+    validateBlankString(faultName, "fault name");
+    Fault fault = operation.getFault(faultName);
+    validateNotNull(fault, "The fault name [" + faultName + "] was not found in the current wsdl file.");
+    return fault;
+  }
+
+  public Definition getDefinition() {
+    return definition;
+  }
+
+  public Set<String> getSchemas() {
+    return schemaCollector.getSchemas(definition);
+  }
+
+  public Service getService() {
+    return service;
+  }
+
+  public Port getPort() {
+    return port;
+  }
+
+  /**
+   * Given a Wsdl location (either local or remote) it will fetch the definition. If the definition cannot be created, then
+   * an exception will be raised
+   *
+   * @param wsdlLocation path to the desired WSDL file
+   */
+  private Definition parseWsdl(final String wsdlLocation) {
+    try {
+      validateBlankString(wsdlLocation, "Wsdl Location");
+
+      WSDLFactory factory = WSDLFactory.newInstance();
+      ExtensionRegistry registry = factory.newPopulatedExtensionRegistry();
+      registry.registerSerializer(Types.class,
+                                  new QName("http://www.w3.org/2001/XMLSchema", "schema"),
+                                  new SchemaSerializer());
+
+      // these will replace whatever may have already been registered
+      // in these places, but there's no good way to check what was
+      // there before.
+      QName header = new QName("http://schemas.xmlsoap.org/wsdl/soap/", "header");
+      registry.registerDeserializer(MIMEPart.class,
+                                    header,
+                                    registry.queryDeserializer(BindingInput.class, header));
+      registry.registerSerializer(MIMEPart.class,
+                                  header,
+                                  registry.querySerializer(BindingInput.class, header));
```

Comment:
from `newPopulatedExtensionRegistry` to here => `ExtensionRegistry registry = initializeRegistry(factory)` (initialize/create/get/setup...)